### ISSUE TITLE
Comments in array indexes interpreted as computed properties?

### ISSUE THREAD

**Issue by dandv** [eslint/eslint#7630]
**Tell us about your environment**

* **ESLint Version:** 3.10.2
* **Node Version:** 6.9.1
* **npm Version:** 3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```js
module.exports = {
    "extends": "standard",
    "installedESLint": true,
    "plugins": [
        "standard",
        "promise"
    ]
};
```
**What did you do? Please include the actual source code causing the issue.**

```js
let a = []
a[1/* the first element is the foo */] = 2
```

**What did you expect to happen?**
No warnings or errors

**What actually happened? Please include the actual, raw output from ESLint.**
2:1  error  Expected 1 or 0 spaces around "[" and "]"  standard/computed-property-even-spacing

**Comment by not-an-aardvark** [comment]
Hi, thanks for the report.

It looks like the error is being reported by the `standard/computed-property-even-spacing` rule, which is a custom rule and isn't part of ESLint itself. You should file an issue on [the repo that maintains that rule](https://github.com/xjamundx/eslint-plugin-standard) instead.

**Comment by dandv** [comment]
I've tried reproducing this with by extending `google`, but I get an odd error:

```
module.exports = {
    "extends": "google",
    "installedESLint": true
};
```

1:5  error  Parsing error: Unexpected token a

**Comment by not-an-aardvark** [comment]
@dandv It looks like the error occurs because you're using `let`, but ESLinton doesn't support ES6 by default. To enable ES6, you can add `es6: true` to your config file:

```js
module.exports = {
  // ...other config options
  env: {
    es6: true
  }
};

**Comment by not-an-aardvark** [comment]
I'm closing this issue because it looks like it was not a problem with ESLint itself. However, feel free to continue the discussion.
